### PR TITLE
docs: document color functions

### PR DESCRIPTION
**Pull request by Gravy59**
The first of many?

### CODE REVIEW

**Comment by Gravy59** [comment]
Please squash these commits!

File: docs/color.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+Found in `lua/pixelui/core/cl_color.lua`
+
+# PIXEL.DecToHex
+
+```lua
+function PIXEL.DecToHex(dec, zeros)
```
Comment (TomDotBat) [inline docs/color.md:6]: If we're documenting the full library then it'd be worth moving this to a more appropriate section such as "Utilities". This is fine for now however.

File: docs/color.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+- color (Color) - The color created from the hexadecimal string.
+
+---
+
+# PIXEL.GetRainbowColor
+
+```lua
+function PIXEL.GetRainbowColor()
+```
+
```
Comment (TomDotBat) [inline docs/color.md:127]: Keep the structure of method documentation consistent.

File: docs/color.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+```lua
+function PIXEL.DecToHex(dec, zeros)
+```
+
+Converts a decimal value to a hexadecimal string representation.
+
+**Parameters:**
+- `dec` (number) - The decimal value to convert.
+- `zeros` (number, optional) - The number of leading zeros to add to the hexadecimal string. Defaults to 2.
```
Comment (TomDotBat) [inline docs/color.md:13]: Maybe use italics for optional parameters? I'll let you decide what looks best.

File: docs/color.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+**Returns:**
+- `hex` (string) - The hexadecimal string representation of the decimal value.
+
+---
+
+# PIXEL.ColorToHex
+
+```lua
+function PIXEL.ColorToHex(color)
+```
```
Comment (TomDotBat) [inline docs/color.md:24]: Perhaps we can make this a bit more concise:
```markdown
# ColorToHex

```lua
PIXEL.ColorToHex(color)
```
The document outline would look cleaner without several occurrences of "PIXEL.". Additionally, I don't think the function keyword is necessary, however if you have reasoning for this please say so.

File: docs/color.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+# PIXEL.HSLToColor
+
+```lua
+function PIXEL.HSLToColor(h, s, l, a)
+```
+
+Converts HSL (Hue, Saturation, Lightness) values to a color.
+
+**Parameters:**
+- `h` (number) - The hue value.
```
Comment (TomDotBat) [inline docs/color.md:63]: It would be better to use ordered lists for method parameters. The same can be done for returns that have multiple values.

**Review by TomDotBat — CHANGES_REQUESTED** [review]
Looks good so far, just a few formatting related concerns.

**Comment by TomDotBat** [comment]
Relates to #29

**Comment by TomDotBat** [comment]
Make this PR on: https://github.com/TomDotBat/docs

**Pull request closed without merging**